core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java

```diff
@@ -0,0 +1,583 @@
+        sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'after'})",
+                Map.of("name", name, "query", query));
+        awaitTriggerDiscovered(db, name, query);
+
+
+        db.executeTransactionally("MATCH (a:A {name: \"A\"})-[r:R2]->(z:Z {name: \"Z\"})\n" +
+                "DELETE r");
+
+        assertEventually(() ->
+                        db.executeTransactionally("MATCH (a:AA) RETURN a", Map.of(),
+                                result -> {
+                                    final Node r = result.<Node>columnAs("a").next();
+                                    return (boolean) r.getProperty("triggerAfter", false)
+                                            && r.getProperty("deleted", "").equals("R2");
+                                })
+                , (value) -> value, 30L, TimeUnit.SECONDS);
+    }
+    
+    //
+    // new test cases
+    //
+
+    @Test
+    public void testTriggerShow() {
+        String name = "test-show1";
+        String name2 = "test-show2";
+        String query = "MATCH (c:TestShow) SET c.count = 1";
+        
+        testCall(sysDb, "CALL apoc.trigger.install('neo4j', $name, $query,{}) YIELD name",
+                map("query", query, "name", name),
+                r -> assertEquals(name, r.get("name")));
+        
+        testCall(sysDb, "CALL apoc.trigger.install('neo4j', $name, $query,{}) YIELD name",
+                map("query", query, "name", name2),
+                r -> assertEquals(name2, r.get("name")));
+        
+        // not updated
+        testResult(sysDb, "CALL apoc.trigger.show('neo4j') " +
+                        "YIELD name, query RETURN * ORDER BY name",
+                map("query", query, "name", name),
+                res -> {
+                    Map<String, Object> row = res.next();
+                    assertEquals(name, row.get("name"));
+                    assertEquals(query, row.get("query"));
+                    row = res.next();
+                    assertEquals(name2, row.get("name"));
+                    assertEquals(query, row.get("query"));
+                    assertFalse(res.hasNext());
+                });
+    }
+    
+    @Test
+    public void testInstallTriggerInUserDb() {
+        try {
+            testCall(db, "CALL apoc.trigger.install('neo4j', 'userDb', 'RETURN 1',{})",
+                    r -> fail("Should fail because of user db execution"));
+        } catch (QueryExecutionException e) {
+            assertTrue(e.getMessage().contains(TRIGGER_NOT_ROUTED_ERROR));
+        }
+    }
+
+    @Test
+    public void testInstallTriggerInSystemDb() {
+        try {
+            testCall(sysDb, "CALL apoc.trigger.install('system', 'name', 'SHOW DATABASES', {})", r -> fail(""));
+        } catch (RuntimeException e) {
+            assertTrue(e.getMessage().contains(TRIGGER_BAD_TARGET_ERROR));
+        }
+    }
+
```

Comment:
Could we add a test where we start the dbms, then create a database "foo", then installs a trigger for "foo", and then check that the trigger actually works?

This is because we weren't sure whether whether the transaction event listeners would get registered for the database "foo" if it was created after the dbms starts up.

Hint: it should work fine, but I'd like a test to make it continues to work fine into the future

Reply:
This is another one of those tests that might be good to back-port to 4.4 if you agree?

Reply:
Yes, it would be good to port it to 4.4. 

As above, I think it's better to use `TriggerClusterRoutingTest` or create another docker enterprise test

Reply:
Added to `TriggerEnterpriseFeaturesTest`